### PR TITLE
absolutize tool paths for (configure_)make (#593)

### PR DESCRIPTION
**Pull request by jheaff1**
This change is necessary if a hermetic cc toolchain is being used
(e.g. gcc checked into the workspace repository)

Testing: Successfully built zlib using configure_make and a custom
hermetic cc toolchain.

### CODE REVIEW

**Comment by UebelAndre** [comment]
Seems good to me, can you think of any way we could prevent this from regressing in future changes? Would hate to break your use of the rules again in the future 😅

**Review by UebelAndre — APPROVED** [review]

**Comment by jheaff1** [comment]
> Seems good to me, can you think of any way we could prevent this from regressing in future changes? Would hate to break your use of the rules again in the future 😅

That's a good question. Does rules_foreign_cc have any automated tests as part of its CI pipeline? If so, could there be some tests that use a hermetic cc toolchain?

**Comment by UebelAndre** [comment]
> That's a good question. Does rules_foreign_cc have any automated tests as part of its CI pipeline? If so, could there be some tests that use a hermetic cc toolchain?

The examples are all run as part of CI: https://github.com/bazelbuild/rules_foreign_cc/tree/main/examples

Perhaps something could be added there? Or if there's a easy way to go setup a hermetic toolchain under some condition we could enable that in a dedicated job for building these targets.

**Comment by jheaff1** [comment]
@UebelAndre That could definitely be achieved. Here are a few projects that define cc toolchains:

* https://github.com/grailbio/bazel-toolchain
* https://github.com/silvergasp/bazel-embedded
* https://github.com/d-asnaghi/bazel-arm-none-eabi
* 

One of these projects could be used by rules_foreign_cc. Or perhaps it’s better that rules_foreign_cc define its own toolchain, to prevent a cyclic dependency of e.g bazel-embedded depends on rules_foreign_cc which depends on bazel-embedded

File: foreign_cc/private/make_script.bzl [hunk 1]
```diff
@@ -142,7 +142,8 @@ def _get_configure_variables(tools, flags, user_env_vars):
     for tool in _CONFIGURE_TOOLS:
         tool_value = getattr(tools, _CONFIGURE_TOOLS[tool])
         if tool_value:
-            tools_dict[tool] = [tool_value]
+            # Force absolutize of tool paths, which may relative to the workspace dir (hermetic toolchains) be provided in project repositories (i.e hermetic toolchains)
```
Comment (jheaff1) [inline foreign_cc/private/make_script.bzl:145]: Need to fix the wording of this comment

**Pull request merged**